packages/teams-js/test/private/files.spec.ts

```diff
@@ -548,54 +550,123 @@ describe('files', () => {
+        utils.initializeWithContext('content');
 
-    it('should not allow calls without frame context initialization', async () => {
-      await utils.initializeWithContext('settings');
-      expect(() => files.openDownloadFolder(null, emptyCallback)).toThrowError(
-        'This call is only allowed in following contexts: ["content"]. Current context: "settings"',
-      );
-    });
+        const callback = (err: SdkError): void => {
+          try {
+            expect(err).toBeFalsy();
+            done();
+          } catch (error) {
+            done(error);
+          }
+        };
 
-    // null file path value is interpreted as opening cofigured download preference folder
-    it('should send the message to parent correctly with file path as null', () => {
-      utils.initializeWithContext('content');
+        files.openDownloadFolder(null, callback);
 
-      const callback = jest.fn(err => {
-        expect(err).toBeFalsy();
+        const openDownloadFolderMessage = utils.findMessageByFunc('files.openDownloadFolder');
+        expect(openDownloadFolderMessage).not.toBeNull();
+        utils.respondToMessage(openDownloadFolderMessage, false);
       });
 
-      files.openDownloadFolder(null, callback);
+      // non-null file path value is interpreted as opening containing folder for the given file path
+      it('should send the message to parent correctly with non-null file path', done => {
+        utils.initializeWithContext('content');
+        const sendAndHandleSdkErrorSpy = jest.spyOn(communication, 'sendAndHandleSdkError');
+
+        const callback = jest.fn(err => {
+          try {
+            expect(err).toBeFalsy();
+            done();
+          } catch (error) {
+            done(error);
+          }
+        });
+
+        files.openDownloadFolder('fileObjectId', callback);
 
-      const openDownloadFolderMessage = utils.findMessageByFunc('files.openDownloadFolder');
-      expect(openDownloadFolderMessage).not.toBeNull();
-      utils.respondToMessage(openDownloadFolderMessage, false);
-      expect(callback).toHaveBeenCalled();
+        const openDownloadFolderMessage = utils.findMessageByFunc('files.openDownloadFolder');
+        expect(openDownloadFolderMessage).not.toBeNull();
+        utils.respondToMessage(openDownloadFolderMessage, false);
+        expect(sendAndHandleSdkErrorSpy).toHaveBeenCalled();
+        // expect(callback).toHaveBeenCalled();
+      });
     });
+    describe('v2', () => {
+      it('should not allow calls before initialization', async () => {
+        expect(() => files.openDownloadFolder(null)).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.keys(FrameContexts)
+        .map(k => FrameContexts[k])
+        .forEach(frameContext => {
+          it(`should not allow calls from ${frameContext} context`, async () => {
+            if (frameContext === FrameContexts.content) {
+              return;
```

Comment:
Unit Test case guideline states we avoid the use of `return`, reference code : `Object.keys(FrameContexts).forEach(k => {
      const context = FrameContexts[k];
      if (allowedContexts.some(allowedContext => allowedContext === context)) {`
      add test cases in if block.